artifactory/commands/transferfiles/utils.go

```diff
@@ -342,3 +350,52 @@ func addErrorToChannel(errorsChannelMng *ErrorsChannelMng, file FileUploadStatus
 	}
 	return false
 }
+
+// Stop transferring if one of the following happened:
+// * Error occurred while handling errors (for example - not enough space in file system)
+// * Error occureed during delayed artifacts handling
+// * User interrupted the process (ctrl+c)
+func ShouldStop(phase *phaseBase, delayHelper *delayUploadHelper, errorsChannelMng *ErrorsChannelMng) bool {
+	if phase != nil && phase.shouldStop() {
+		log.Debug("Stop transferring data - Interrupted.")
+		return true
+	}
+	if delayHelper != nil && delayHelper.delayedArtifactsChannelMng.shouldStop() {
+		log.Debug("Stop transferring data - error occurred while handling transfer's delayed artifacts files.")
+		return true
+	}
+	if errorsChannelMng != nil && errorsChannelMng.shouldStop() {
+		log.Debug("Stop transferring data - error occurred while handling transfer's errors.")
+		return true
+	}
+	return false
+}
+
+func getRunningNodes(sourceRtDetails *coreConfig.ServerDetails) ([]string, error) {
+	serviceManager, err := utils.CreateServiceManager(sourceRtDetails, retries, retriesWait, false)
+	if err != nil {
+		return nil, err
+	}
+	return serviceManager.GetRunningNodes()
+}
+
+func stopAllRunningNodes(srcUpService *srcUserPluginService, runningNodes []string) {
```

Comment:
stopAllRunningNodes --> stopTransferOnArtifactoryNodes

(less of a scary name :) Let's rename the test corresponding test function as well).